components/availability/src/Availability.svelte

```diff
@@ -171,20 +177,81 @@
     }
   };
 
+  // Consecutive same-availability periods of time, from earliest start_time to latest end_time.
+  const generateEpochs = (
+    slots: SelectableSlot[],
+    participant_threshold: number,
+  ) => {
+    let scale = scaleTime().domain([
+      slots[0].start_time,
+      slots[slots.length - 1].end_time,
+    ]);
+    let epochs = slots
+      .reduce((m, n) => {
+        if (
+          m[m.length - 1] &&
+          JSON.stringify(m[m.length - 1][0].available_calendars) ===
+            JSON.stringify(n.available_calendars)
+        ) {
+          m[m.length - 1].push(n);
+        } else {
+          m.push([n]);
+        }
+        return m;
+      }, [] as TimeSlot[][])
+      .map((epoch) => {
```

Comment:
It would be a bit more complicated, but I think the work being done by this `map` can be done by the `reduce` instead to save iterating over the epochs again. Basically, we would update the properties of the previous epoch's `end_time`, `height`, and `slots` instead of `m[m.length - 1].push(n)`.